staramr/detection/AMRDetection.py

```diff
@@ -177,14 +179,223 @@ def run_amr_detection(self, files, pid_threshold, plength_threshold_resfinder, p
             self._pointfinder_dataframe = self._create_pointfinder_dataframe(pointfinder_blast_map, pid_threshold,
                                                                              plength_threshold_pointfinder, report_all)
 
-        self._summary_dataframe = self._create_amr_summary(files, self._resfinder_dataframe,
+        self._summary_dataframe = self._create_amr_summary(files, self._resfinder_dataframe,self._quality_module_dataframe,
                                                            self._pointfinder_dataframe, self._plasmidfinder_dataframe, self._mlst_dataframe)
 
-        self._detailed_summary_dataframe = self._create_detailed_amr_summary(files, self._resfinder_dataframe,
+        self._detailed_summary_dataframe = self._create_detailed_amr_summary(files, self._resfinder_dataframe,self._quality_module_dataframe,
                                                                              self._pointfinder_dataframe,
                                                                              self._plasmidfinder_dataframe,
                                                                              self._mlst_dataframe)
 
+    def _create_quality_module_dataframe(self,files):
+        name_set=[]
+        for myFile in files:
+            name_set.append(path.splitext(path.basename(myFile))[0])
+        files_genome_lengths = self._get_genome_lengths(files)
+        files_genome_length_feedback = self._get_genome_length_feedback(files_genome_lengths)
+        files_contigs_lengths=self._get_files_contigs_lengths(files)
+        files_N50_values=self._get_N50(files_contigs_lengths,files_genome_lengths)
+        files_N50_feedback=self._get_N50_feedback(files_N50_values)
+        minimum_contig_length=1000
+        files_contigs_under_minimum_bp= self._get_num_contigs_under_minimum_bp(files_contigs_lengths,minimum_contig_length)
+        unacceptable_num_contigs_under_minimum_bp= 3
```

Comment:
Both minimum_contig_length and unacceptable_num_contigs_under_minimum_bp should be variabled passed to function with default.